### ISSUE TITLE
Lightbox doesn't display when `<amp-img>` is wrapped in `<a>` 

### ISSUE THREAD

**Issue by kienstra** [ampproject/amphtml#25021]
## What's the issue?

It looks like when `<amp-img>` elements that have an `<amp-image-lightbox>` are wrapped in `<a>`, the lightbox doesn't open on clicking the image. 

Though maybe this is the expected behavior. The [amp-image-lightbox](https://amp.dev/documentation/components/amp-image-lightbox/) documentation doesn't seem to mention `<a>`.

## How do we reproduce the issue?
https://codepen.io/ryankienstra/pen/WNNrrOJ

1. In an AMP document, add an `<amp-img on="tap:amp-image-lightbox">` with the required attributes, including `src`
2. Add an [amp-image-lightbox](https://amp.dev/documentation/components/amp-image-lightbox/) custom script to the `<head>`
3. Add this element below the closing `</body>`:
```html
<amp-image-lightbox id="amp-image-lightbox" layout="nodisplay" data-close-button-aria-label="Close"></amp-image-lightbox>
```
4. Wrap the `<amp-img>` in `<a href="https://example.com">`
5. Expected: on clicking the image, a lightbox should open
6. Actual: it goes to the URL of the `<a>`, instead of opening a lightbox
7. Remove the `<a>`, and observer that the lightbox opens on clicking the image. 

![amp-image-lightbox-reproduction](https://user-images.githubusercontent.com/4063887/66690705-88074b00-ec57-11e9-824e-96399fa74f7e.gif)

There doesn't seem to be a way to keep the `<a>`, even by moving the `on="tap:amp-image-lightbox"` to be an attribute of it.

## What browsers are affected?
Desktop (macOS 10.13.6)
Chrome 77.0.3865.90 (Official Build) (64-bit)
Firefox 69.0.1 (64-bit)
Safari 11.1.2 (13605.3.8)

## Which AMP version is affected?
1910071803120

This was reported on October 4th, 2019 in our work on the AMP Plugin for WordPress, in https://github.com/ampproject/amp-wp/issues/3450

**Comment by westonruter** [comment]
cc @cathyxz

**Comment by cathyxz** [comment]
I think this is currently the expected behavior for `amp-image-lightbox`. We're actually in process of deprecating `amp-image-lightbox` and would encourage using the `amp-lightbox-gallery` component for lightboxing images instead. 

/cc @nainar @sparhami for `amp-image-lightbox` deprecation.

**Comment by kienstra** [comment]
Hi @cathyxz,
Thanks for looking at this. I'll work on using `amp-lightbox-gallery` instead.

**Comment by kienstra** [comment]
**Question about `amp-lightbox-gallery`**

Hi @cathyxz,
Thanks a lot for your help with this.

Is the `amp-lightbox-gallery` supposed to be able to have its `<amp-img>` wrapped in `<a>`? It looks like the same issue appears for this.

**Steps to reproduce**
1. Go to the [amp-lightbox-gallery Playground](https://playground.amp.dev/?url=https%3A%2F%2Fpreview.amp.dev%2Fdocumentation%2Fexamples%2Fcomponents%2Famp-lightbox-gallery&format=websites)
2. Wrap an existing `<amp-img>` that has a `lightbox` attribute in an `<a>`, like:

```html
<a href="https://example.com/here">
    <amp-img lightbox src="/static/samples/img/road-1.jpg" width="300" height="200" layout="responsive"></amp-img>
</a>
```
3. Reload the preview panel
4. Click on the image
5. Expected: the lightbox opens
6. Actual: it goes to the `href` of the `<a>` that it's wrapped in

![amp-image-gallery](https://user-images.githubusercontent.com/4063887/66783190-0c96db00-ee9d-11e9-95ed-fe00bf46cd7b.gif)

Thanks, Cathy!

**Comment by cathyxz** [comment]
I'm curious about your use case, in my impression, most times you wrap content in a `<a>`, it *is* supposed to navigate to the url that it's in. One valid use case I can think about for not having that behavior is when you're using your `<a>` tag as an anchor, so you expect the lightbox to open, while the rest of the page scrolls to your anchor under the lightbox. Is that your use case, or are you trying to do something different?

**Comment by kienstra** [comment]
Hi @cathyxz,
Sure, the use case is an Image block in WordPress, where it's possible to:

1. Select a link (`<a>`) to wrap the image in. If this is in Transitional mode, this will apply to the AMP and non-AMP URLs.
2. Select 'lightbox,' and if this is in Transitional mode, this will only apply to the AMP URL.

<img width="1436" alt="wp-editor-image-block" src="https://user-images.githubusercontent.com/4063887/66786467-06592c80-eea6-11e9-9a93-bb6e98d4f4b8.png">

It's possible that users want the `<a>` to work on the non-AMP URL in Transitional mode, but also want the lightbox to work on the AMP URL.

>One valid use case I can think about for not having that behavior is when you're using your <a> tag as an anchor, so you expect the lightbox to open, while the rest of the page scrolls to your anchor under the lightbox. Is that your use case, or are you trying to do something different?

No, that's not the use case I had in mind 😄 

Thanks, Cathy!

**Comment by nainar** [comment]
Hi @kienstra,

Sorry I am trying to understand your issue. In the non AMP page the user should click on the image and go to the href and the AMP page should go to the lightbox? Why would you want the AMP and non AMP page to behave differently?

**Comment by kienstra** [comment]
Hi @nainar,
Thanks for looking at this!

>In the non AMP page the user should click on the image and go to the href and the AMP page should go to the lightbox? 

Yes

>Why would you want the AMP and non AMP page to behave differently?

This is for the [AMP Plugin for WordPress](https://github.com/ampproject/amp-wp), so the plugin doesn't modify the non-AMP behavior. 

The 'Add lightbox effect' toggle here can only affect the AMP URL, as it's outside the scope of the plugin to change the non-AMP URL:

<img width="1436" alt="wp-editor-image-block" src="https://user-images.githubusercontent.com/4063887/66791021-ec284a00-eeb7-11e9-92fa-2ac3031a9d83.png">

**Comment by cathyxz** [comment]
Summarizing an offline discussion with @sparhami (thank you for looking into this!): 

From a technical/implementation perspective, this boils down to whether or not tap actions should `preventDefault`. 

Under the hood in lightbox, we register an `activate` method on a `tap` action: 
https://github.com/ampproject/amphtml/blob/5cf2e94646aeb5d8dfb51db74ac8084b9ab6aa0d/extensions/amp-lightbox-gallery/0.1/service/lightbox-manager-impl.js#L286

Per the below code in the `action-impl.js` library, we preventDefault for keyboard interactions, but not for `tap` actions. 
https://github.com/ampproject/amphtml/blob/af35a6eae86e38933dff9f1d2c9f24a30e915a0d/src/service/action-impl.js#L303-L328

We could enable the behavior that @kienstra is proposing by adding `preventDefault` in the actions library, but I'm concerned that doing this would be a major API break, since we currently do not `preventDefault`. 

Alternatively, we could directly register click actions for lightboxed images in `lightbox-manager-impl` and add a `preventDefault` there. This would still break backwards compatibility though, just on a slightly smaller scale. However, given the auto-lightbox feature, would still merit quite a bit of discussion (e.g. should we avoid auto-lightboxing images that are in `<a>` tags? Should we avoid preventDefault-ing them?). 

In short, although implementing this feature is not incredibly difficult, whether to implement this, how to treat tap actions (preventDefault or not) in lightbox, auto-lightbox, and in general, merits a longer discussion. I think the current behavior is the expected behavior in most cases, but I can be convinced otherwise if there is a good argument for better end user experience.

**Comment by kienstra** [comment]
Hi @cathyxz,
Thanks a lot for working on this.

Right, this would break backwards-compatibility. I'm still not sure if `<amp-img lightbox>` should be able to be wrapped in `<a>`, with the lightbox still working.

Thanks a lot for your help here.

**Comment by cathyxz** [comment]
I think the conclusion right now is no and we can't add this feature due to backwards compatibility, sorry.

**Comment by kienstra** [comment]
Sure, that's fine 😄
